maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/report/StatelessXmlReporter.java

```diff
@@ -240,6 +240,10 @@ public void testSetCompleted( WrappedReportEntry testSetReportEntry, TestSetStat
             }
             ppw.endElement(); // TestSuite
         }
+        catch ( final IOException e )
+        {
+            throw new ReporterException( "Failure generating XML report.", e );
```

Comment:
@ChristianSchulte 
Pls do not commit. I have stash with explanation.
Later. First of all is to test freebasd.
I have fixed testng system properties in a branch but have not commit it to the branch. We will test it. But later in the evening.